client/protocols/unlock.py

```diff
@@ -153,12 +153,14 @@ def interact(self, unique_id, case_id, question_prompt, answer, choices=None, ra
                 correct = True
             tg_id = -1
             misU_count_dict = {}
+            rationale = "Unknown"
 
             if not correct:
-                misU_count_dict, tg_id, printed_msg = self.guidance_util.show_guidance_msg(unique_id,input_lines,
-                    self.hash_key)
-
+                guidance_data = self.guidance_util.show_guidance_msg(unique_id, input_lines,
+                                                                     self.hash_key)
+                misU_count_dict, tg_id, printed_msg, rationale = guidance_data
             else:
+                rationale = self.guidance_util.prompt_with_prob()
```

Comment:
@ksteph - How often should we be prompting on correct answers? Currently it's set to a 50% probability on the first correct answer in a session and then backs off by cutting the probability in half every time. 

That's almost certainly way too often?